### PR TITLE
Implementation of an amplitudes printer

### PR DESCRIPTION
**Pull request by mfherbst**
This PR implements a `describe_amplitude` function for `ExcitedStates`, where Tensor indices can be formatted in various ways (using HOMO/LUMO and similar labels, the adcc index' convention or the host programm's index convention). Some more generalisation and cleanup of the code is needed.

### CODE REVIEW

**Comment by mfherbst** [comment]
@maxscheurer This can go in now. This PR introduces a couple of classes, which can be used for formatting tensor indices in general. This way one can also get a good overview over other kind of tensors (e.g. ERI elements, density matrix elements, ...) and display the tensor indices in various ways (adcc indexing, HF provider indexing, relative to HOMO / LUMO etc.).

File: adcc/MoSpaces.py [hunk 1]
```diff
@@ -20,79 +20,157 @@
...
 
 import libadcc
 
+from .backends import import_scf_results
+from .memory_pool import memory_pool
+
 __all__ = ["MoSpaces"]
 
 
+def expand_spaceargs(hfdata, **spaceargs):
```
Comment (maxscheurer) [inline adcc/MoSpaces.py]: I guess this was moved here from the C-side?

Comment (maxscheurer) [reply adcc/MoSpaces.py]: Never mind... I saw now that it came from `ReferenceState`...

Comment (mfherbst) [reply adcc/MoSpaces.py]: Yes indeed. It came from `ReferenceState`, where it was a duplicate from `C`-side for reasons I cannot remember, so good it's now only once at an understandable place.

**Pull request merged**